File: frontend/src/types/organization.ts
/** 用户角色 */
export const RoleMap: Record<string, string> = {
  admin: '管理员',
  manager: '部门经理',
  finance: '财务',
  viewer: '查看者',
}
export type UserRole = 'admin' | 'manager' | 'finance' | 'viewer'

/** 部门 */
export interface DeptItem {
  id: number
  name: string
  manager: string
  memberCount: number
  totalBudget: number
}

/** 用户 */
export interface UserItem {
  id: number
  username: string
  realName: string
  deptName: string
  role: UserRole
  status: string
}

/** 部门表单 */
export interface DeptForm {
  name: string
  manager: string
}

/** 用户表单 */
export interface UserForm {
  username: string
  realName: string
  deptId: number | null
  role: UserRole
}

File: backend/app/schemas/organization.py
"""组织权限 Schema（与前端 types/organization.ts 对齐）"""
from pydantic import BaseModel, Field


class DeptCreate(BaseModel):
    name: str = Field(..., description="部门名称")
    manager: str = Field("", description="负责人")


class DeptUpdate(BaseModel):
    name: str | None = None
    manager: str | None = None


class DeptResponse(BaseModel):
    id: int
    name: str
    manager: str = ""
    member_count: int = 0
    total_budget: float = 0.0


class UserCreate(BaseModel):
    username: str = Field(..., description="用户名")
    real_name: str = Field("", description="真实姓名")
    dept_id: int | None = None
    role: str = Field("viewer", description="角色")


class UserUpdate(BaseModel):
    real_name: str | None = None
    dept_id: int | None = None
    role: str | None = None


class UserResponse(BaseModel):
    id: int
    username: str
    real_name: str = ""
    dept_name: str = ""
    role: str = "viewer"
    status: str = "active"